annotation-file-utilities/src/annotator/find/Insertion.java

```diff
@@ -50,22 +50,66 @@
     protected Set<String> packageNames;
 
     /**
+     * The inner types to go on this insertion. See {@link ReceiverInsertion}
+     * for more details.
+     */
+    protected List<Insertion> innerTypeInsertions;
+
+    /**
      * Set of annotation names that should always be qualified, even
      *  when {@link getText(boolean, boolean)} is called with abbreviate true.
      */
     protected static Set<String> alwaysQualify = new LinkedHashSet<String>();
 
     /**
+     * Type to be inserted, if applicable; otherwise null.
+     */
+    protected Type type;
+
+    /**
+     * If true only the annotations from {@link type} will be inserted.
+     */
+    protected boolean annotationsOnly;
+
+    /**
      * Creates a new insertion.
      *
      * @param criteria where to insert the text
      * @param separateLine whether to insert the text on its own
      */
     public Insertion(Criteria criteria, boolean separateLine) {
+        this(null, criteria, separateLine, null);
+    }
+
+    /**
+     * Creates a new insertion.
+     *
+     * @param type type to be generated, if applicable
+     * @param criteria where to insert the text
+     * @param innerTypeInsertions insertions to go on inner parts of type
+     */
+    public Insertion(Type type, Criteria criteria,
+            List<Insertion> innerTypeInsertions) {
+        this(type, criteria, false, innerTypeInsertions);
+    }
+
+    /**
+     * Creates a new insertion.
+     *
+     * @param type type to be generated, if applicable
```

Comment:
I meant "generated" in the sense of source code generation.  I suppose "inserted" would be clearer here.